google-cloud-pubsublite/src/test/java/com/google/cloud/pubsublite/it/ITVPCNegativeTest.java

```diff
@@ -0,0 +1,391 @@
+    } catch (Exception e) {
+      fail(
+          "Expected PERMISSION_DENIED StatusRuntimeException during createTopic but got: "
+              + e.toString());
+    }
+  }
+
+  @Test
+  public void deniedDeleteTopic() {
+    try {
+      adminClient.deleteTopic(topicPath).get();
+      fail("Expected PERMISSION_DENIED StatusRuntimeException");
+    } catch (InterruptedException e) {
+      fail("Expected PERMISSION_DENIED StatusRuntimeException but got: " + e.toString());
+    } catch (ExecutionException e) {
+      Throwable thrown = e.getCause();
+      checkExceptionForVPCError((StatusRuntimeException) thrown);
+    }
+  }
+
+  @Test
+  public void deniedGetTopic() {
+    try {
+      adminClient.getTopic(topicPath).get();
+      fail("Expected PERMISSION_DENIED StatusRuntimeException");
+    } catch (InterruptedException e) {
+      fail("Expected PERMISSION_DENIED StatusRuntimeException but got: " + e.toString());
+    } catch (ExecutionException e) {
+      Throwable thrown = e.getCause();
+      checkExceptionForVPCError((StatusRuntimeException) thrown);
+    }
+  }
+
+  @Test
+  public void deniedListTopics() {
+    try {
+      adminClient.listTopics(locationPath).get();
+      fail("Expected PERMISSION_DENIED StatusRuntimeException");
+    } catch (InterruptedException e) {
+      fail("Expected PERMISSION_DENIED StatusRuntimeException but got: " + e.toString());
+    } catch (ExecutionException e) {
+      Throwable thrown = e.getCause();
+      checkExceptionForVPCError((StatusRuntimeException) thrown);
+    }
+  }
+
+  @Test
+  public void deniedListTopicSubscriptions() {
+    try {
+      adminClient.listTopicSubscriptions(topicPath).get();
+      fail("Expected PERMISSION_DENIED StatusRuntimeException");
+    } catch (InterruptedException e) {
+      fail("Expected PERMISSION_DENIED StatusRuntimeException but got: " + e.toString());
+    } catch (ExecutionException e) {
+      Throwable thrown = e.getCause();
+      checkExceptionForVPCError((StatusRuntimeException) thrown);
+    }
+  }
+
+  @Test
+  public void deniedUpdateTopic() {
+    try {
+      FieldMask mask = FieldMask.newBuilder().addPaths("partition_config.scale").build();
+      adminClient.updateTopic(topic, mask).get();
+      fail("Expected PERMISSION_DENIED StatusRuntimeException");
+    } catch (InterruptedException e) {
+      fail("Expected PERMISSION_DENIED StatusRuntimeException but got: " + e.toString());
+    } catch (ExecutionException e) {
+      Throwable thrown = e.getCause();
+      checkExceptionForVPCError((StatusRuntimeException) thrown);
+    }
+  }
+
+  @Test
+  public void deniedCreateSubscription() {
+    try {
+      adminClient.createSubscription(subscription).get();
+    } catch (InterruptedException e) {
+      fail("Expected PERMISSION_DENIED StatusRuntimeException but got: " + e.toString());
+    } catch (ExecutionException e) {
+      Throwable thrown = e.getCause();
+      checkExceptionForVPCError((StatusRuntimeException) thrown);
```

Comment:
Have you ever run this? Most of the exceptions thrown by the client libs are StatusException not StatusRuntimeException. Use ExtractStatus.toCanonical here and elsewhere to not have to care about this.

Reply:
Yes in most cases I received `StatusRuntimeException`s except when calling Publish and Subscribe.
However opting to use `ExtractStatus.toCanonical` as it is much cleaner.